### ISSUE TITLE
Cannot reproduce quantitative result on the StereoMIS dataset

### ISSUE THREAD

**Issue by nhphucqt** [lastbasket/Endo-4DGS#17]
Hi, thank you for your excellent work. Currently I am trying to reproduce the quantitative result on the StereoMIS data. I downloaded the dataset from https://zenodo.org/records/8154924 and ran the `prepare_stereomis.sh` script. My dataset folder is prepared as follows:

```
stereomis
|-- split_1
      |-- images
           |-- 000801.png
           |-- ...
           |-- 001000.png
      |-- poses_bounds.npy
      |-- depth_dam (generated from scripts/pre_dam_dep.py)
           |-- 000801.npy
           |-- ...
           |-- 001000.npy
      |-- masks (copy from the StereoMIS_0_0_1/P1)
           |-- 000241l.png
           |-- ...
           |-- 016799l.png
```

Then I ran the following commands:

```bash
python train.py -s /volume/data/stereomis/split_1 --port 6017 --expname "stereomis/split_1" --configs arguments/stereomis.py 
python render.py --model_path output/stereomis/split_1 --pc --skip_video --skip_train --configs arguments/stereomis.py
python metrics.py --model_path output/stereomis/split_1
```

The quantitative result I obtained is:

```json
{
  "ours_6000": {
    "SSIM": 0.8504340648651123,
    "PSNR": 31.88469696044922,
    "LPIPS": 0.18058928847312927
  }
}
```

However, this is different from the results reported in the paper:

```json
{
  "ours_6000": {
    "SSIM": 0.8509106636047363,
    "PSNR": 32.695579528808594,
    "LPIPS": 0.14808064699172974
  }
}
```

# Code modification for StereoMIS

To run the method on the StereoMIS dataset, I made a small modification in the dataset loader.

In scene/endo_loader.py, I changed the depth folder name from the original implementation (lines 229–232):

https://github.com/lastbasket/Endo-4DGS/blob/39c6ce22d4f53b86cf8455f985ebaf65959c0462/scene/endo_loader.py#L229-L232

to:

```python
self.depth_paths = agg_fn("depth_dam")
```

# Questions

1. Could you confirm whether the dataset folder structure above is correct (especially the mask naming and indexing)?
2. For StereoMIS, are the masks expected to match the train split frame IDs exactly (e.g., 000801.png -> 000801l.png), or is the mask folder supposed to be pre-filtered per split?

Thanks a lot for your help.

**Comment by nhphucqt** [comment]
I also visualized the output point cloud. Below is what I get for `frame_0`:

<img height="250" alt="Image" src="https://github.com/user-attachments/assets/eef81a2e-1812-4825-9c26-69170d278a37" />
<img height="250" alt="Image" src="https://github.com/user-attachments/assets/7b69e995-4e55-4640-a4d9-b67f9509cc1b" />

However, this looks different from the pretrained Gaussians you provided:

<img height="250" alt="Image" src="https://github.com/user-attachments/assets/72e2884f-2c1c-4e7e-a9c6-e7fb5204821f" />
<img height="250" alt="Image" src="https://github.com/user-attachments/assets/2627af57-b8e5-4013-badf-b3aa5ead2000" />

Could you please help me understand what might be causing this discrepancy?

**Comment by 007ARSENALZHQ** [comment]
Hi, I’m running Endo-4DGS on StereoMIS and I’m seeing a similar issue.
[ITER 6000] Evaluating test: L1 0.01670101739685325 PSNR 31.411726558909695
it  different from the results reported in the paper
dataset from P1  000801.png to 001000.png
Did you manage to resolve this, and if so, what was the root cause?

**Comment by nhphucqt** [comment]
> Hi, I’m running Endo-4DGS on StereoMIS and I’m seeing a similar issue. [ITER 6000] Evaluating test: L1 0.01670101739685325 PSNR 31.411726558909695 it different from the results reported in the paper dataset from P1 000801.png to 001000.png Did you manage to resolve this, and if so, what was the root cause?

Hi, I found an issue in the convention conversion script:

https://github.com/lastbasket/Endo-4DGS/blob/39c6ce22d4f53b86cf8455f985ebaf65959c0462/scene/endo_loader.py#L206

The StereoMIS dataset camera convention is already in OpenCV/COLMAP format, but the script above converts it to OpenGL convention. I think the convention should just be kept unchanged.

```python
poses = np.concatenate([poses[..., :1], poses[..., 1:2], poses[..., 2:3], poses[..., 3:4]], -1)
```

Also, regarding the output point cloud: `pre_dam_dep.py` uses `depth_anything_vitb14.onnx`, while the paper used `depth_anything_vits14.onnx`. When I replaced `depth_anything_vitb14.onnx` with `depth_anything_vits14.onnx`, the resulting point cloud looked the same as the author’s checkpoint.

Hope this helps 👍

**Comment by 007ARSENALZHQ** [comment]
> > 您好，我在 StereoMIS 上运行 Endo-4DGS 时遇到了类似的问题。[ITER 6000] 评估测试：L1 0.01670101739685325 PSNR 31.411726558909695，这与论文数据集中 P1 000801.png 到 001000.png 的结果不同。请问您是否解决了这个问题？如果解决了，根本原因是什么？
> 
> 您好，我发现约定转换脚本存在问题：
> 
> [Endo-4DGS/scene/endo_loader.py](https://github.com/lastbasket/Endo-4DGS/blob/39c6ce22d4f53b86cf8455f985ebaf65959c0462/scene/endo_loader.py#L206)
> 
> 第 206 行 [39c6ce2](/lastbasket/Endo-4DGS/commit/39c6ce22d4f53b86cf8455f985ebaf65959c0462)
> 
>  poses  =  np.concatenate ( [ poses [..., : 1 ], -poses [..., 1 : 2 ] , -poses [ ..., 2 : 3 ], poses [ ... , 3 : 4 ]], -1 )
> StereoMIS 数据集的相机命名规范已经是 OpenCV/COLMAP 格式，但上面的脚本将其转换为 OpenGL 格式。我认为应该保持命名规范不变。
> 
> poses = np.concatenate([poses[..., :1], poses[..., 1:2], poses[..., 2:3], poses[..., 3:4]], -1)
> 另外，关于输出点云：`pre_dam_dep.py`我使用的是`depth_anything_vitb14.onnx`，而论文中使用的是`depth_anything_vits14.onnx`。当我将替换`depth_anything_vitb14.onnx`为后`depth_anything_vits14.onnx`，生成的点云看起来与作者的检查点相同。
> 
> 希望这能帮到你👍

Thank you for your reply. At the moment, I have already resolved most of the issues. I would like to ask how you handled the missing masks for the even-numbered frames in the dataset. Additionally, which frames did you use to obtain the final PSNR reported when reproducing the results in the paper?
